datapackage_pipelines/lib/join.py

```diff
@@ -207,7 +223,7 @@ def new_resource_iterator(resource_iterator_):
 def process_target_resource(source_spec, resource):
     target_fields = \
         resource.setdefault('schema', {}).setdefault('fields', [])
-    added_fields = sorted(fields.keys())
+    added_fields = sorted(fields.keys()) if not maintain_fields_order else fields.keys()
```

Comment:
fields.keys() does not guarantee to return keys in order of insertion.
see https://docs.python.org/3.6/tutorial/datastructures.html#dictionaries

it's makes more sense to me to convert the dictionary form to the list form, and use that form throughout the code, instead of the other way round.